src/test/java/com/google/research/bleth/simulator/ObserverTest.java

```diff
@@ -0,0 +1,930 @@
+package com.google.research.bleth.simulator;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.runner.RunWith;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.Mock;
+
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ObserverTest {
+    private static final Location zeroOnZeroCoordinate = new Location(0, 0);
+    private static final Location oneOnOneCoordinate = new Location(1, 1);
+    private static final Location zeroOnOneCoordinate = new Location(0, 1);
+    private static final Location oneOnZeroCoordinate = new Location(1, 0);
+    private static final BeaconFactory beaconFactory = new BeaconFactory();
+    private static final ObserverFactory observerFactory = new ObserverFactory();
+
+    @Mock
+    private Simulation simulation;
+
+    /**
+     * A fake implementation for IResolver that allows to test the arguments it receives when an observer calls receiveInformation.
+     */
+    private class FakeResolver implements IResolver {
```

Comment:
Consider splitting this to a separate file (even if it's not going to be reused - just so it doesn't clutter the tests file).